client/src/hooks/useUser.js

```diff
@@ -0,0 +1,49 @@
+import { useRouter } from 'next/router'
+import React from 'react'
+import api from '../api'
+import { ResourcesPortalContext } from '../ResourcesPortalContext'
+import { useLocalStorage } from './useLocalStorage'
+
+export const useUser = (defaultUser, defaultToken, redirectUrl) => {
+  const [loginRedirectUrl, setLoginRedirectUrl] = useLocalStorage(
+    'redirectUrl',
+    redirectUrl
+  )
+  const { user, setUser, token, setToken } = React.useContext(
+    ResourcesPortalContext
+  )
+  const router = useRouter()
+  React.useEffect(() => {
+    if (defaultUser && defaultToken) {
+      // Component cannot update other component may be coming from here
+      setUser(defaultUser)
+      setToken(defaultToken)
+      if (loginRedirectUrl) {
+        setLoginRedirectUrl()
+        router.replace(loginRedirectUrl)
+      }
+    }
+  })
+  const isLoggedIn = Boolean(user && token)
+  const refreshUserData = () => {
```

Comment:
I think this needs to be async and we need to await the api calls

Reply:
Cool, done.